File: php/connect.php
<?php
$database = 'test';

$manager = new MongoDB\Driver\Manager(
  "mongodb://CN=ChrisChoClient,OU=TestClientCertificateOrgUnit,O=TestClientCertificateOrg,L=TestClientCertificateLocality,ST=TestClientCertificateState,C=US@localmongo1/?ssl=true&authMechanism=MONGODB-X509",
  [],
  [
    "pem_file" => "/Users/ccho/dev/drivers/certs/v3/test-client.pem",
    "ca_file" => "/Users/ccho/dev/drivers/certs/v3/test-ca.pem",
  ]
);


$bulk = new MongoDB\Driver\BulkWrite;
$bulk->insert(['phptest' => 'ok']);

$manager->executeBulkWrite('test.stuff', $bulk);

File: ruby/connect.rb
require 'mongo'

begin
  client = Mongo::Client.new( 
                             ['localmongo1:27017'],
                             auth_mech: :mongodb_x509,
                             ssl: true,
                             ssl_cert:    '/Users/ccho/dev/x509-driver-snippets/certs/test-client.pem',
                             ssl_key:    '/Users/ccho/dev/x509-driver-snippets/certs/test-client.pem',
                             ssl_ca_cert: '/Users/ccho/dev/x509-driver-snippets/certs/test-ca.pem',

                            )
                             #ssl_verify: false,
                             #tlsInsecure: true)

  puts Thread.current.backtrace

  db = client.database[:test]
  collection = client[:stuff]

  doc = { name: 'Ruby', hobbies: [ 'hiking', 'tennis', 'fly fishing' ] }

  result = collection.insert_one(doc)
  puts result.n
rescue StandardError => e
  puts e.backtrace
end

